play/src/front/Rules/PersonalAreaPropertyEditorRules/PersonalAreaPropertyEditorRules.ts

```diff
@@ -0,0 +1,102 @@
+import { PersonalAreaAccessClaimMode, PersonalAreaPropertyData } from "@workadventure/map-editor";
+import { derived, get, Readable } from "svelte/store";
+import { mapEditorSelectedAreaPreviewStore } from "../../Stores/MapEditorStore";
+
+export type Option = {
+    value: string;
+    label: string;
+    created: undefined | boolean;
+};
+
+export class PersonalAreaPropertyEditorRules {
```

Comment:
This class feels weird to me. It is only instantiated in the `PersonalAreaPropertyEditor.svelte` component.
It does not expect any constructor argument so we could directly export functions and the store.

+ if we expose the store directly in Svelte, we can use the "$store" notation and not worry about subscribing to it.

The thing with Svelte is that as long as you are in a Svelte file, you get reactivity, which makes the code much easier to write. As soon as you want to put your code in a TS file, you have to deal with stores manually and it is harder.
The upcoming Svelte 5 will fix this with Runes, but we are now only in Svelte 3.

As it is now, I believe we should refactor this class. I understand the need for a level of abstraction but it is probably not the good one. The number of `get(someStore)` also seems to indicate we should be dealing with stores at the Svelte level.

Reply:
The PersonalAreaPropertyEditorRule is removed.